This file is part of mingw-cross-env.
See doc/index.html for further information.

Contains ad hoc patches for cross building.

commit 637620b33c175676972419ae674806825a390734
Author: Mark Brand <mabrand@mabrand.nl>
Date:   Thu Sep 23 21:36:04 2010 +0200

    no gtk doc check

diff --git a/configure.ac b/configure.ac
index a821267..2e541d2 100644
--- a/configure.ac
+++ b/configure.ac
@@ -2741,8 +2741,6 @@ dnl **************************
 dnl *** Checks for gtk-doc ***
 dnl **************************
 
-GTK_DOC_CHECK([1.15])
-
 AC_ARG_ENABLE(man,
               [AC_HELP_STRING([--enable-man],
                               [regenerate man pages from Docbook [default=no]])],enable_man=yes,

commit 8f4fde8f0ab605720ef0baf7ef8080479b34af63
Author: Mark Brand <mabrand@mabrand.nl>
Date:   Thu Sep 23 21:41:51 2010 +0200

    optional gettext

diff --git a/configure.ac b/configure.ac
index 2e541d2..18c2dea 100644
--- a/configure.ac
+++ b/configure.ac
@@ -464,8 +464,8 @@ AC_SUBST([CONFIG_STATUS_DEPENDENCIES],['$(top_srcdir)/po/LINGUAS'])
 GLIB_GNU_GETTEXT
 
 if test "$gt_cv_have_gettext" != "yes" ; then
-  AC_MSG_ERROR([
-*** You must have either have gettext support in your C library, or use the 
+  AC_MSG_WARN([
+*** You should have either have gettext support in your C library, or use the
 *** GNU gettext library. (http://www.gnu.org/software/gettext/gettext.html
 ])
 fi
diff --git a/glib/gi18n.h b/glib/gi18n.h
index c710046..269bfdf 100644
--- a/glib/gi18n.h
+++ b/glib/gi18n.h
@@ -22,7 +22,12 @@
 
 #include <glib.h>
 
+#ifdef ENABLE_NLS
 #include <libintl.h>
+#else
+#define gettext(String) (String)
+#endif
+
 #include <string.h>
 
 #define  _(String) gettext (String)

commit 0a998b24fdb3ae5397ff5ad498260165300a4393
Author: Mark Brand <mabrand@mabrand.nl>
Date:   Thu Sep 23 21:42:46 2010 +0200

    fix tool paths

diff --git a/glib-2.0.pc.in b/glib-2.0.pc.in
index 3d1ac12..58e1ecf 100644
--- a/glib-2.0.pc.in
+++ b/glib-2.0.pc.in
@@ -3,9 +3,9 @@ exec_prefix=@exec_prefix@
 libdir=@libdir@
 includedir=@includedir@
 
-glib_genmarshal=glib-genmarshal
-gobject_query=gobject-query
-glib_mkenums=glib-mkenums
+glib_genmarshal=@bindir@/glib-genmarshal
+gobject_query=@bindir@/gobject-query
+glib_mkenums=@bindir@/glib-mkenums
 
 Name: GLib
 Description: C Utility Library

commit 7ddbe18ccce7aede0cef972c6035058151c3ce10
Author: Mark Brand <mabrand@mabrand.nl>
Date:   Thu Sep 23 21:45:50 2010 +0200

    fix dependency libs

diff --git a/configure.ac b/configure.ac
index 18c2dea..e472768 100644
--- a/configure.ac
+++ b/configure.ac
@@ -2698,7 +2698,7 @@ case $host in
 	G_LIBS_EXTRA="-luser32 -lkernel32"
     ;;
   *-*-mingw*)
-	G_LIBS_EXTRA="-lws2_32 -lole32"
+	G_LIBS_EXTRA="-lws2_32 -lole32 -lshlwapi"
     ;;
   *)
 	G_LIBS_EXTRA=""
diff --git a/glib-2.0.pc.in b/glib-2.0.pc.in
index 58e1ecf..17b2133 100644
--- a/glib-2.0.pc.in
+++ b/glib-2.0.pc.in
@@ -11,5 +11,5 @@ Name: GLib
 Description: C Utility Library
 Version: @VERSION@
 Libs: -L${libdir} -lglib-2.0 @INTLLIBS@
-Libs.private: @ICONV_LIBS@
+Libs.private: @G_THREAD_LIBS@ @G_LIBS_EXTRA@ @PCRE_LIBS@ @INTLLIBS@ @ICONV_LIBS@
 Cflags: -I${includedir}/glib-2.0 -I${libdir}/glib-2.0/include @GLIB_EXTRA_CFLAGS@
